controllers/spiaccesscheck_controller.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/redhat-appstudio/service-provider-integration-operator/pkg/spi-shared/config"
+
+	"github.com/redhat-appstudio/service-provider-integration-operator/pkg/serviceprovider"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	api "github.com/redhat-appstudio/service-provider-integration-operator/api/v1beta1"
+)
+
+// SPIAccessCheckReconciler reconciles a SPIAccessCheck object
+type SPIAccessCheckReconciler struct {
+	client.Client
+	Scheme                 *runtime.Scheme
+	ServiceProviderFactory serviceprovider.Factory
+	Configuration          config.Configuration
+}
+
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=spiaccesschecks,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=spiaccesschecks/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=spiaccesschecks/finalizers,verbs=update
+
+func (r *SPIAccessCheckReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	lg := log.FromContext(ctx)
+
+	ac := api.SPIAccessCheck{}
+	if err := r.Get(ctx, req.NamespacedName, &ac); err != nil {
+		if errors.IsNotFound(err) {
+			lg.Info("SPIAccessCheck not found on cluster",
+				"namespace:name", fmt.Sprintf("%s:%s", req.Namespace, req.Name))
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, NewReconcileError(err, "failed to load the SPIAccessCheck from the cluster")
+	}
+
+	if ac.Status.Ttl != 0 {
+		if time.Now().After(time.Unix(ac.Status.Ttl, 0)) {
```

Comment:
What about not storing the `Ttl` in the status and just basing this decision on the difference between `time.Now()` and `ac.ObjectMeta.CreationTimestamp`?

Reply:
yes, fixed